### PR TITLE
[WIP] splitting conda-builds into separate build and test phases for PRs

### PR DESCRIPTION
**Pull request by pjh5**

### CODE REVIEW

**Comment by pjh5** [comment]
Use this jenkins job https://ci.pytorch.org/jenkins/job/caffe2-pull-request/856/

File: .jenkins/test.sh [hunk 1]
```diff
@@ -14,6 +14,13 @@ fi
 
 # The prefix must mirror the setting from build.sh
 INSTALL_PREFIX="/usr/local/caffe2"
+
+# Anaconda builds have a special install prefix and python
+if [[ "$BUILD_ENVIRONMENT" == conda* ]]; then
```
Comment (orionr) [inline .jenkins/test.sh:19]: Does this need another * at the front of conda?

File: .jenkins/test.sh [hunk 2]
```diff
@@ -26,8 +33,12 @@ if [[ "${BUILD_ENVIRONMENT}" == *-android* ]]; then
   exit 0
 fi
 
-export PYTHONPATH="${PYTHONPATH}:${INSTALL_PREFIX}/${SITE_DIR}"
-export LD_LIBRARY_PATH="${LD_LIBRARY_PATH}:${INSTALL_PREFIX}/lib"
+# Set PYTHONPATH and LD_LIBRARY_PATH so that python can find the installed
+# Caffe2. This shouldn't be done on Anaconda, as Anaconda should handle this.
+if [[ "$BUILD_ENVIRONMENT" != conda* ]]; then
```
Comment (orionr) [inline .jenkins/test.sh:38]: Also - do you want *conda*?

**Review by orionr — APPROVED** [review]
Looks reasonable so long as you're okay with building without testing. I'm guessing that is by design, though.

**Pull request merged**